File: app/src/main/java/com/killain/organizer/packages/interactors/DBInjectorModule.java
package com.killain.organizer.packages.interactors;

import android.content.Context;

import com.killain.organizer.packages.database.AppDatabase;
import com.killain.organizer.packages.interfaces.SubTaskDAO;
import com.killain.organizer.packages.interfaces.TaskDAO;

import dagger.Module;
import dagger.Provides;

@Module
public class DBInjectorModule {

    private AppDatabase db;
    private TaskDAO taskDAO;
    private Context context;
    private SubTaskDAO subTaskDAO;

    public DBInjectorModule(Context context) {
        this.context = context;
        db = AppDatabase.getAppDatabase(context);
        taskDAO = db.getTaskDAO();
        subTaskDAO = db.getSubTaskDAO();
    }

    @Provides
    public AppDatabase getDb() {
        return db;
    }

    @Provides
    public TaskDAO getTaskDAO() {
        return taskDAO;
    }

    @Provides
    public SubTaskDAO getSubTaskDAO() {
        return subTaskDAO;
    }
}

File: app/src/main/java/com/killain/organizer/packages/interactors/DBInjectorModule.kt
package com.killain.organizer.packages.interactors

import android.content.Context

import com.killain.organizer.packages.database.AppDatabase
import com.killain.organizer.packages.interfaces.SubTaskDAO
import com.killain.organizer.packages.interfaces.TaskDAO

import dagger.Module
import dagger.Provides
import org.jetbrains.annotations.Nullable

@Module
class DBInjectorModule(context: Context) {

    val db: AppDatabase = AppDatabase.getAppDatabase(context)!!
        @Provides
        get() = field

    val taskDAO: TaskDAO
        @Provides
        get() = field

    val subTaskDAO: SubTaskDAO
        @Provides
        get() = field

    init {
        taskDAO = db.taskDAO
        subTaskDAO = db.subTaskDAO
    }

//    @Provides
//    fun getDb() : AppDatabase? {
//        return db
//    }

//    @Provides
//    fun getTaskDao() : TaskDAO? {
//        return taskDAO
//    }
//
//    @Provides
//    fun getSubTaskDAO() : SubTaskDAO? {
//        return subTaskDAO
//    }
}
